app.py

```diff
@@ -920,6 +920,21 @@ def main_page(page):
         else:
             return utils.page_403(TRANSLATIONS, current_user()['username'], g.lang, TRANSLATIONS.get_translations(g.lang, 'ui').get('not_user'))
 
+    user = current_user()
+
+    if page == 'for-teachers':
+        for_teacher_translations = hedyweb.PageTranslations(page).get_page_translations(g.lang);
+        if is_teacher(user):
+            welcome_teacher = session.get('welcome-teacher') or False
+            session['welcome-teacher'] = False
```

Comment:
```suggestion
            session.pop('welcome-teacher', None)
```

Remove the value from the session so we don't need to store it. Sessions get transmitted over the wire (in cookies)

Reply:
Didn't know that! Code was already there and it seemed important, will change right away.